vividus-plugin-rest-api/src/main/java/org/vividus/archive/steps/ArchiveSteps.java

```diff
@@ -106,22 +112,43 @@ public void saveArchiveEntriesToVariables(DataWrapper archiveData, List<ArchiveV
     public void verifyArchiveContainsEntries(DataWrapper archiveData, List<NamedEntry> parameters)
     {
         Set<String> entryNames = ZipUtils.readZipEntryNamesFromBytes(archiveData.getBytes());
-
+        MutableBoolean attachmentLoaded = new MutableBoolean(false);
         parameters.forEach(entry ->
         {
             String expectedName = entry.getName();
             if (entry.getRule() != null)
             {
                 StringComparisonRule comparisonRule = entry.getRule();
-                softAssert.assertThat(String.format(
-                        "The archive contains entry matching the comparison rule '%s' with name pattern '%s'",
-                        comparisonRule, expectedName), entryNames, hasItem(comparisonRule.createMatcher(expectedName)));
+                if (!softAssert.assertThat(String.format(
+                        "The archive contains entry matching the comparison "
+                        + "rule '%s' with name pattern '%s'", comparisonRule, expectedName), entryNames,
+                        hasItem(comparisonRule.createMatcher(expectedName))))
+                {
+                    attachmentLoaded.setValue(true);
+                }
             }
             else
             {
-                softAssert.assertThat("The archive contains entry with name " + expectedName, entryNames,
-                        hasItem(expectedName));
+                if (!softAssert.assertThat("The archive contains entry with name " + expectedName,
+                        entryNames, hasItem(expectedName)))
+                {
+                    attachmentLoaded.setValue(true);
+                }
             }
         });
+        if (entryNames.isEmpty())
+        {
+            publishAttachment("There are no files in the archive");
+            attachmentLoaded.setValue(false);
+        }
+        if (attachmentLoaded.isTrue())
+        {
+            publishAttachment(entryNames.toString());
```

Comment:
its going to be comma separated list of strings which is not readable, please consider listing them in HTML format using for example table

Reply:
done